packages/protocol/contracts/layer1/provers/ProverSet.sol

```diff
@@ -94,67 +94,49 @@ contract ProverSet is EssentialContract, IERC1271 {
         LibAddress.sendEtherAndVerify(admin, _amount);
     }
 
-    /// @notice Proposes a batch blocks only when it is the first batch blocks proposal in the
-    /// current L1 block.
-    function proposeBlocksV2Conditionally(
-        bytes[] calldata _params,
-        bytes[] calldata _txList
-    )
-        external
-        onlyProver
-    {
-        ITaikoL1 taiko = ITaikoL1(taikoL1());
-        // Ensure this block is the first block proposed in the current L1 block.
-        require(taiko.lastProposedIn() != block.number, NOT_FIRST_PROPOSAL());
-        taiko.proposeBlocksV2(_params, _txList);
-    }
-
-    /// @notice Propose a Taiko block.
-    function proposeBlockV2(bytes calldata _params, bytes calldata _txList) external onlyProver {
-        ITaikoL1(taikoL1()).proposeBlockV2(_params, _txList);
-    }
-
     /// @notice Propose multiple Taiko blocks.
-    function proposeBlocksV2(
-        bytes[] calldata _paramsArr,
-        bytes[] calldata _txListArr
+    function proposeBlocksV3(
```

Comment:
Shall we keep `proposeBlocksV2` or using `ForkManager` in `ProverSet`? Otherwise, we need to deploy another `ProverSet` and deposit tokens into it?

Reply:
This is indeed an outstanding issue. Please test this new protocol without using ProverSet, I'll create another PR later.